docs/content/guides/dagster/managing-ml.mdx

```diff
@@ -0,0 +1,296 @@
+
+Dagster’s <PyObject object="MetadataValue" /> types enable types such as tables, URLs, notebooks, Markdown, etc. In the following example, the Markdown metadata type is used to generate plots. Each plot will show a specific evaluation metric’s performance throughout each training iteration also known as an epoch during the training cycle.
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=ui_plot_start   endbefore=ui_plot_end
+from dagster import MetadataValue
+import seaborn
+import matplotlib.pyplot as plt
+import base64
+from io import BytesIO
+
+
+def make_plot(eval_metric):
+    plt.clf()
+    training_plot = seaborn.lineplot(eval_metric)
+    fig = training_plot.get_figure()
+    buffer = BytesIO()
+    fig.savefig(buffer)
+    image_data = base64.b64encode(buffer.getvalue())
+    return MetadataValue.md(f"![img](data:image/png;base64,{image_data.decode()})")
+```
+
+In this example, we’re using a dictionary called metadata to store the Markdown plots and the score value in Dagster.
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=metadata_use_start endbefore=metadata_use_end
+from dagster import asset
+import xgboost as xgb
+from sklearn.metrics import mean_absolute_error
+
+
+@asset
+def xgboost_comments_model(transformed_training_data, transformed_test_data):
+    transformed_X_train, transformed_y_train = transformed_training_data
+    transformed_X_test, transformed_y_test = transformed_test_data
+    # Train XGBoost model, which is a highly efficient and flexible model
+    xgb_r = xgb.XGBRegressor(
+        objective="reg:squarederror", eval_metric=mean_absolute_error, n_estimators=20
+    )
+    xgb_r.fit(
+        transformed_X_train,
+        transformed_y_train,
+        eval_set=[(transformed_X_test, transformed_y_test)],
+    )
+
+    ## plot the mean absolute error values as the training progressed
+    metadata = {}
+    for eval_metric in xgb_r.evals_result()["validation_0"].keys():
+        metadata[f"{eval_metric} plot"] = make_plot(
+            xgb_r.evals_result_["validation_0"][eval_metric]
+        )
+    # keep track of the score
+    metadata["score (mean_absolute_error)"] = xgb_r.evals_result_["validation_0"][
+        "mean_absolute_error"
+    ][-1]
+
+    return Output(xgb_r, metadata=metadata)
+```
+
+If we look at the `xgboost_comments_model` asset in the Dagster UI, we see the metadata rendered. Numerical values, such as the `score (mean_absolute_error)` will be logged and plotted for each materialization, which can be useful to understand the score over time for machine learning models.
+
+<Image
+alt="alt"
+src="/images/guides/managing_ml/managing_ml_ui.png"
+width={1188}
+height={541}
+/>
+
+The Markdown plots are also available to inspect the evaluation metrics during the training cycle by clicking on **\[Show Markdown]**. <Image
+alt="alt"
+src="/images/guides/managing_ml/plot_ui.png"
+width={1188}
+height={541}
+/>
```

Comment:
```suggestion
The Markdown plots are also available to inspect the evaluation metrics during the training cycle by clicking on **\[Show Markdown]**:

<Image
alt="alt"
src="/images/guides/managing_ml/plot_ui.png"
width={1188}
height={541}
/>
```

The image tag wrapped to the preceding line, causing some weird spacing issues when rendered.